src/NLog/Targets/FileTarget.cs

```diff
@@ -1831,11 +1831,7 @@ private void TryArchiveFile(string fileName, LogEventInfo ev, int upcomingWriteS
 #endif
                 try
                 {
-                    archiveFile = this.GetArchiveFileName(fileName, ev, upcomingWriteSize);
-                    if (!string.IsNullOrEmpty(archiveFile))
-                    {
-                        this.DoAutoArchive(archiveFile, ev);
-                    }
+                    this.DoAutoArchive(archiveFile, ev);
```

Comment:
It seems ok. Because already created a file name at line 1815 and already checked if it is null or empty at line 1816. If file name is null, then already this line will not executed.